### PR TITLE
ci: add service UDS tokenizer tests

### LINKED ISSUE #306: Add UDS tokenizer tests to CI
**Description**

The UDS tokenizer is now part of the main execution path following the isolation of embedded tokenizers behind build tags (#295, #303). However, we have no automated testing in CI to validate the UDS tokenizer service itself. This creates a significant gap in our test coverage for production code.

**Suggested Plan**

- **Add CI workflow for UDS tokenizer tests:**
    - Set up python environment
    - Run pytest unit and integration tests
- Add test coverage for edge cases
- **Update integration tests to use the pytest testing framework:** Convert `run_integration_tests.py` into a proper pytest suite with fixtures for server lifecycle management (setup/teardown), clean state between tests, etc.

### PR DESCRIPTION
**Pull request by sagearc**
Adds CI testing for the UDS tokenizer service with GitHub Actions integration. This PR updates the test infrastructure to use `pytest` to handle UDS server lifecycle (initialization and teardown), with more extensive tests coverage.

- All previous tests are covered in `test_integration.py` and more
- Server lifecycle is managed through pytest fixtures in `conftest.py`
- Package dependencies (build/test) and source code is unified in `pyproject.toml`

closes #306

### CODE REVIEW

File: services/uds_tokenizer/tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        yield socket_path
+
+
+@pytest.fixture(scope="session")
+def tokenizer_service(uds_socket_path: str) -> Iterator[TokenizerService]:
+    """Provide the TokenizerService instance used by the gRPC server."""
+    service = TokenizerService()
+    thread_pool = get_thread_pool()
+    server = create_grpc_server(service, uds_socket_path, thread_pool)
+    server.start()
```
[the remark this answers is not in the document]
Comment (sagearc) [reply services/uds_tokenizer/tests/conftest.py:57]: added `wait_for_ready`

File: services/uds_tokenizer/tests/test_integration.py [hunk 2]
```diff
@@ -0,0 +1,327 @@
+# Copyright 2025 The llm-d Authors.
```
Comment (vMaroon) [inline services/uds_tokenizer/tests/test_integration.py]: new files should have 2026 boilerplate, relevant to all greens in this PR.
